src/unix/thread.c

```diff
@@ -72,10 +73,85 @@ int uv_thread_create(uv_thread_t *tid, void (*entry)(void *arg), void *arg) {
 }
 
 
+#if (defined(__APPLE__) && defined(__MACH__)) || defined(_AIX)
+int uv_thread_cpumasksize(void) {
+  return -ENOTSUP;
+}
+
+int uv_thread_setaffinity(uv_thread_t* tid,
+                          char* cpumask,
+                          char* oldmask,
+                          size_t mask_size) {
+  return -ENOTSUP;
+}
+
+
+int uv_thread_getaffinity(uv_thread_t* tid,
+                          char* cpumask,
+                          size_t mask_size) {
+  return -ENOTSUP;
+}
+
+#else /* !((defined(__APPLE__) && defined(__MACH__)) || defined(_AIX)) */
+
+int uv_thread_cpumasksize(void) {
+  return CPU_SETSIZE;
+}
+
+int uv_thread_setaffinity(uv_thread_t* tid,
+                          char* cpumask,
+                          char* oldmask,
+                          size_t mask_size) {
+  int i;
+  int r;
+  cpu_set_t cpuset;
+  int cpumasksize = uv_thread_cpumasksize();
```

Comment:
Style nit: libuv style is to declare variables upfront, and assign them afterwards.